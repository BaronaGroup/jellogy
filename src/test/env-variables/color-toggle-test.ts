import {prepareTestKarhu} from '../test-utils'

describe('default-log-level-test', () => {
  const karhuTest = prepareTestKarhu({})

  it('allows overriding default log level', karhuTest((karhu, output) => {
    process.env.KARHU_COLOR = 'true'
    karhu.context('test-context').error('Should have colors')
    process.env.KARHU_COLOR = 'false'
    karhu.context('test-context').error('Should not have colors')
    expect(output.tracked).toMatchSnapshot()
  }))
})
