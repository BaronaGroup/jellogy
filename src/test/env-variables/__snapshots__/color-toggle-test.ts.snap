// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default-log-level-test allows overriding default log level 1`] = `
Map {
  "NONE" => Array [],
  "DEBUG" => Array [],
  "INFO" => Array [],
  "WARN" => Array [],
  "ERROR" => Array [
    Array [
      "0 ERROR test-context[97m[101m",
      "Should have colors",
      "[49m[39m",
    ],
    Array [
      "1 ERROR test-context",
      "Should not have colors",
    ],
  ],
}
`;
